fe2-android/app/src/main/java/com/github/dedis/popstellar/di/KeysetModule.java

```diff
@@ -56,14 +58,25 @@ public static AndroidKeysetManager provideDeviceKeysetManager(
       Ed25519PrivateKeyManager.registerPair(true);
       PublicKeySignWrapper.register();
 
-      // TODO: move to background thread
-
-      return new AndroidKeysetManager.Builder()
-          .withSharedPref(applicationContext, DEVICE_KEYSET_NAME, DEVICE_SHARED_PREF_FILE_NAME)
-          .withKeyTemplate(Ed25519PrivateKeyManager.rawEd25519Template())
-          .withMasterKeyUri(DEVICE_MASTER_KEY_URI)
-          .build();
-    } catch (IOException | GeneralSecurityException e) {
+      CompletableFuture<AndroidKeysetManager> future =
+          CompletableFuture.supplyAsync(
+              () -> {
+                try {
+                  return new AndroidKeysetManager.Builder()
+                      .withSharedPref(
+                          applicationContext, DEVICE_KEYSET_NAME, DEVICE_SHARED_PREF_FILE_NAME)
+                      .withKeyTemplate(KeyTemplates.get("ED25519_RAW"))
+                      .withMasterKeyUri(DEVICE_MASTER_KEY_URI)
+                      .build();
+                } catch (GeneralSecurityException | IOException e) {
+                  throw new SecurityException(e);
+                }
+              },
+              Executors.newSingleThreadExecutor());
+
+      return future.join();
```

Comment:
I see the point but then I should change the signature to return a Completable rather than the `AndroidKeysetManager` and as it's injected I'm not very sure I'm allowed to